### ISSUE TITLE
[imgui] Link errors on gl calls

### ISSUE THREAD

**Issue by r0levrai** [microsoft/vcpkg#27104]
While installing imgui with `vcpkg/vcpkg install imgui[core,sdl2-binding,opengl3-binding,docking-experimental]`, and trying to run the [corresponding example from upstream docking branch](https://github.com/ocornut/imgui/blob/docking/examples/example_sdl_opengl3/main.cpp) by copying it to an empty `main.cpp` in visual studio with integration set up using `vcpkg/vcpkg integrate install` and build mode set to `x64`, I get three linking errors related to the `glViewport`, `glClearColorand` `glClear` calls from the example:

``````
Severity	Code	Description	File	Line	Suppression State
Error	LNK2019	unresolved external symbol __imp_glClearColor referenced in function SDL_main	[...]\main.obj	1	
Error	LNK2019	unresolved external symbol __imp_glClear referenced in function SDL_main	[...]\main.obj	1	
Error	LNK2019	unresolved external symbol __imp_glViewport referenced in function SDL_main	[...]\main.obj	1	
Error	LNK1120	3 unresolved externals	sdf-test	[...]\x64\Debug\simple-test.exe	1	
``````

Also reproduced on x86-windows building to `x86`, and also without `[docking-experimental]` using the [same example from master branch](https://github.com/ocornut/imgui/blob/master/examples/example_sdl_opengl3/main.cpp).

@JackBoosY, could this be related to #14753 ?

NB:
I had to replace all `#include <SDL_*.h>` by `#include <SDL2/SDL_*.h>` in the examples.
I also originally had one more linking error related to main, which I resolved by manually linking `vcpkg\installed\x64-windows\lib\manual-link\SDL2main.lib` as stated in [the integration readme](https://github.com/microsoft/vcpkg/blob/master/docs/users/buildsystems/msbuild-integration.md) (but it would be cool to have a heads up in the `integrate` or `install` commands log, or maybe in the [quick start guide](https://vcpkg.io/en/getting-started.html)). Leaving it there in case it helps others :)

**Comment by Neumann-A** [comment]
You need to also link `opengl.lib`. Nothing vcpkg could do here to help since the MSBuild integration is flawed by design.

Consider using CMake instead.

**Comment by r0levrai** [comment]
That makes sense, I should have thought about that! Since I red some strange things about opengl, what is the recommended way to get the lib?

I still have a way to go to be able to handcraft a working CMake in a reasonable time, as a c++ noob it's a bit of a learning curve ^^'

**Comment by r0levrai** [comment]
I just installed VS and its c++ extension, I assume it doesn't come with the sdk?
Installing the [windows sdk](https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/) and linking `C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\OpenGL32.Lib` indeed did the trick!!
Thanks a lot, would have never found this by myself.

I just hopped there was a more standard way to do this so that someone can just clone my repo, run a few portable commands and have it running.
Actually, `vcpkg/vcpkg install opengl` seems to work without the sdk nor having to manually link the lib. Should we not have it as a dependency of the `imgui` port?
